src/pytorch_lightning/strategies/sharded_spawn.py

```diff
@@ -93,7 +96,7 @@ def block_backward_sync(self) -> Generator:
             yield None
 
     @rank_zero_only
-    def _optim_state_dict(self, optimizer):
+    def _optim_state_dict(self, optimizer: Optimizer) -> Dict[str, Tensor]:
```

Comment:
[State dict ](https://github.com/pytorch/pytorch/blob/9213751970e4bb4eecd0b5dc6317740f017460a0/torch/optim/optimizer.py#L135)does not return a dict with tensors. 
The documentation [says](https://pytorch.org/docs/stable/generated/torch.optim.Optimizer.state_dict.html#torch.optim.Optimizer.state_dict) "differs between optimizer classes." 
In their [type file](https://github.com/pytorch/pytorch/blob/master/torch/optim/optimizer.pyi), they annotate as Dict[str, Any]. 

Finally, also in the [fairscale OSS optimizer](https://fairscale.readthedocs.io/en/stable/api/optim/oss.html?highlight=OSS#fairscale.optim.OSS), which is used here concretely, they annotate as Dict[str, Any]. 


This is probably the best we can do.

Reply:
Thanks for the suggestions. That makes sense.